app/models/person.rb

```diff
@@ -94,6 +94,71 @@ def downcase_diaspora_handle
       .where(:posts => {:root_guid => StatusMessage.guids_for_author(user.person), :type => 'Reshare'} )
   }
 
+  # This scope selects people where the full name contains the search_str or diaspora ID
+  # starts with the search_str.
+  # However, if the search_str doesn't have more than 1 non-whitespace character, it'll return an empty set.
+  # @param [String] search substring
+  # @return [Person::ActiveRecord_Relation]
+  scope :find_by_substring, ->(search_str) {
+    search_str.strip!
+    return none if search_str.blank? || search_str.size < 2
+
+    sql, tokens = search_query_string(search_str)
+    joins(:profile).where(sql, *tokens)
+  }
+
+  # Left joins likes and comments to a specific post where people are authors of these comments and likes
+  # @param [String, Integer] post ID for which comments and likes should be joined
+  # @return [Person::ActiveRecord_Relation]
+  scope :left_join_visible_post_interactions_on_authorship, ->(post_id) {
+    comments_sql = <<-SQL
+      LEFT OUTER JOIN comments ON
+      comments.author_id = people.id AND comments.commentable_type = 'Post' AND comments.commentable_id = #{post_id}
+    SQL
+
+    likes_sql = <<-SQL
+      LEFT OUTER JOIN likes ON
+      likes.author_id = people.id AND likes.target_type = 'Post' AND likes.target_id = #{post_id}
+    SQL
+
+    joins(comments_sql).joins(likes_sql)
+  }
+
+  # Selects people who can be mentioned in a comment to a specific post. For public posts all people
+  # are allowed, so no additional constraints are added. For private posts selection is limited to
+  # people who have posted comments or likes for this post.
+  # @param [Post] the post for which we query mentionable in comments people
+  # @return [Person::ActiveRecord_Relation]
+  scope :allowed_to_be_mentioned_in_a_comment_to, ->(post) {
+    if post.public?
+      all
+    else
+      left_join_visible_post_interactions_on_authorship(post.id)
+        .where("comments.id IS NOT NULL OR likes.id IS NOT NULL OR people.id = #{post.author_id}")
+    end
+  }
+
+  # This scope adds sorting of people in the order, appropriate for suggesting to a user (current user) who
+  # has requested a list of the people mentionable in a comment for a specific post.
+  # Sorts people in the following priority: post author > commenters > likers > contacts > non-contacts
+  # @param [Post] post for which the mentionable in comment people list is requested
+  # @param [User] user who requests the people list
+  # @return [Person::ActiveRecord_Relation]
+  scope :sort_for_mention_suggestion, ->(post, user) {
+    left_join_visible_post_interactions_on_authorship(post.id)
+      .joins("LEFT OUTER JOIN contacts ON people.id = contacts.person_id AND contacts.user_id = #{user.id}")
+      .joins(:profile)
+      .order(<<-SQL
+        people.id = #{post.author.try(:id).to_i} DESC,
```

Comment:
Oh, I didn't see this before, but this generates this additional query including the search-term:
```sql
SELECT  `people`.* FROM `people` INNER JOIN `profiles` ON `profiles`.`person_id` = `people`.`id` WHERE (`people`.`id` != 73) AND (
      profiles.full_name LIKE '%te%' OR
      people.diaspora_handle LIKE 'te%'
) AND `people`.`id` = 74 LIMIT 1
```
(that's why you needed the `try` here)

So inside a `scope` rails does weird things (it also adds the scope to other queries). When you call `post.author_id` outside the `scope` first (in the `PostService`), it doesn't do that and you don't need a `try`.